apps/dashboard/src/components/conditions-editor/variable-select.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { useMemo, useState, useRef } from 'react';
+import { CheckIcon } from '@radix-ui/react-icons';
+
+import { Code2 } from '@/components/icons/code-2';
+import { Popover, PopoverAnchor, PopoverContent } from '@/components/primitives/popover';
+import { InputPure, InputRoot, InputWrapper } from '@/components/primitives/input';
+import { AUTOCOMPLETE_PASSWORD_MANAGERS_OFF } from '@/utils/constants';
+import { cn } from '@/utils/ui';
+
+const KeyboardItem = ({ children, className }: { children: React.ReactNode; className?: string }) => {
+  return (
+    <span
+      className={cn(
+        'text-foreground-400 shadow-xs text-paragraph-2xs flex h-5 w-5 items-center justify-center rounded-[6px] border border-neutral-200 px-2 py-1 font-light',
+        className
+      )}
+    >
+      {children}
+    </span>
+  );
+};
+
+type VariablesListProps = {
+  options: Array<{ label: string; value: string }>;
+  onSelect: (value: string) => void;
+  selectedValue?: string;
+};
+
+const VariablesList = React.forwardRef<HTMLInputElement, VariablesListProps>(
+  ({ options, onSelect, selectedValue }, ref) => {
+    const [hoveredOptionIndex, setHoveredOptionIndex] = useState(-1);
+    const listRef = useRef<HTMLUListElement>(null);
+
+    const scrollToOption = (index: number) => {
```

Comment:
the list is scrolled when using the keyboard navigation, and the item is not visible in the list